ingestion-sink/Dockerfile

```diff
@@ -1,17 +1,17 @@
 FROM maven:3-jdk-8 AS base
 WORKDIR /app/ingestion-sink
-COPY LICENSE /app/LICENSE
-COPY pom.xml /app/pom.xml
-COPY ingestion-sink/pom.xml /app/ingestion-sink/pom.xml
-COPY checkstyle /app/checkstyle
-COPY ingestion-sink/checkstyle /app/ingestion-sink/checkstyle
+
+FROM base AS build
+COPY LICENSE checkstyle pom.xml /app/
+COPY ingestion-sink/pom.xml ingestion-sink/checkstyle /app/ingestion-sink/
+RUN mvn prepare-package
 COPY ingestion-core/src /app/ingestion-core/src
 COPY ingestion-sink/src /app/ingestion-sink/src
```

Comment:
this now downloads the dependencies before copying `src` directories, so that when building the docker image locally the dependencies are cached, rather than re-downloaded on every source change.